code/modules/fluids/fluid_procs.dm

```diff
@@ -50,16 +50,27 @@ turf/simulated/floor/plating/airless/ocean_canpass()
 	return ..()
 
 
-/turf/proc/fluid_react(var/datum/reagents/R, var/react_volume, var/airborne = 0) //this should happen whenever a liquid reagent hits a simulated tile
+/turf/proc/fluid_react(var/datum/reagents/R, var/react_volume, var/airborne = 0, var/index = 0) //this should happen whenever a liquid reagent hits a simulated tile
 	if (react_volume <= 0) return
 	if (!IS_VALID_FLUIDREACT_TURF(src)) return
+	if (!index)
+		if (airborne)
+			for(var/reagent_id in R.reagent_list)
+				if (ban_from_airborne_fluid.Find(reagent_id)) return
+		else
+			for(var/reagent_id in R.reagent_list)
+				if (ban_from_fluid.Find(reagent_id)) return
+	else // We only care about one chem
+		var/CI = 1
```

Comment:
what is ci

Reply:
It's some grade-A cargo culting; the `/datum/reagent/trans_to_direct` proc uses that variable and structure to pick out the reagent with the right index, when applicable.